modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -78,64 +78,95 @@ class ScaleLayerImpl CV_FINAL : public ScaleLayer
         Mat &outBlob = outputs[0];
         // There is a mode when we multiply a first blob by a second one
         // instead of trainable weights.
-        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();;
+        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();
         Mat bias = hasBias ? (blobs.empty() ? inputs[1] : blobs.back()).reshape(1, 1) : Mat();
 
         MatShape inpShape = shape(inpBlob);
+        const int numSamples = inpShape[0];
+        const int numWeightsSamples = (hasWeights || hasBias) && blobs.empty() ? numSamples : 1;
         const int numWeights = !weights.empty() ? weights.total() : bias.total();
+        const int numWeightsPerSample = numWeights / numWeightsSamples;
```

Comment:
Do we have real example when weights have batch dimension? I mean when we have different NCHW input and NC weights?

Reply:
The scaling parameters can be computed during inference (i.e. untrainable mode). This situation will lead to different weights for different batch items.

EfficientNetB0-YOLO is one example which does it. (https://github.com/opencv/opencv/pull/16436#issuecomment-585732681)

Reply:
issue #17063 also encountered problem in EfficientNetBO ( tensorflow )